calpool-fe/src/pages/SignUp/SignUp.js

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from "react";
+import { Button, TextInput } from "../../components";
+import { Link } from "@mui/material";
+import "./SignUp.css";
+
+const SignUp = () => {
+  const [form, setForm] = useState({
+      firstName: { value: "", error: "" },
+      lastName: { value: "", error: "" },
+      email: { value: "", error: "" },
+      password: { value: "", error: "" },
+  });
+
+  const [showPass, setShowPass] = useState(false);
+
+  const handleFormChange = (event) => {
+    const { name, value } = event.target;
+    setForm(prev => ({...prev, [name]: {value: value, error: ""}}))
+  }
+  
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    const isFormValid = validateForm()
+    if (isFormValid) {
+      //TODO
+    }
+  }
+
+  const validateForm = () => {
```

Comment:
Is there any way you can take these repeated functions like validateForm,  handleFormChange, handleSubmit and make them general by maybe making a utils file or something like that